cli/src/pcluster/cli/middleware.py

```diff
@@ -73,7 +78,11 @@ def update_cluster(func, _body, kwargs):
     if wait and not kwargs.get("dryrun"):
         cloud_formation = boto3.client("cloudformation")
         waiter = cloud_formation.get_waiter("stack_update_complete")
-        waiter.wait(StackName=kwargs["cluster_name"])
+        try:
+            waiter.wait(StackName=kwargs["cluster_name"])
+        except WaiterError as e:
+            LOGGER.error("Failed when waiting for cluster update with error: %s", e)
+            raise APIOperationException(_cluster_status(_body["clusterName"]))
```

Comment:
it looks like this might need to be `_cluster_status(kwargs["cluster_name"])` because the cluster name is provided as a query parameter (which is also what is affecting the naming convention difference here)